backend/src/apiserver/model/pipeline.go

```diff
@@ -39,6 +39,7 @@ type Pipeline struct {
 	// Default version of this pipeline. It could be null.
 	DefaultVersionId string           `gorm:"column:DefaultVersionId;"`
 	DefaultVersion   *PipelineVersion `gorm:"-"`
+	Namespace        string           `gorm:"column:Namespace; size:63; default:''"`
```

Comment:
This PR adds the namespace column but the name column is unique.
Shouldn't we amend this unique index so that the tupple (name, namespace) is unique instead?

Reply:
I'm not sure what you mean here, do we want to make this field unique?

Reply:
That's a good point, we can make the tuple unique.
E.g. two different pipelines can have the same name at different namespaces

Reply:
Thanks @yanniszark, this is quite a big bug :) 

@Bobgy 
By the way, I noticed that we are not using the main gorm repository ```"gorm.io/gorm"```, this will lead to conflict in the future. It should probably become a future refactoring task for the KFP team. 

For example the way to add a composite unique index is shows as below, but this gave me a syntax error with the current gorm repository being used... ```"github.com/jinzhu/gorm"```

```golang
package main

import (
	"gorm.io/driver/mysql"
	"gorm.io/gorm"
)

// ALTER TABLE dev.pipelines ADD UNIQUE `unique_index`(`Name`, `Namespace`);
type Pipeline struct {
	gorm.Model
	Namespace string `gorm:"column:Namespace; size:63; default:''; index:idx_name_ns, unique"`
	Name      string `gorm:"column:Name; not null; index:idx_name_ns"`
}

func main() {
	// db, err := gorm.Open(sqlite.Open("test.db"), &gorm.Config{})
	dsn := "root:@tcp(localhost:3306)/dev?charset=utf8mb4&parseTime=True&loc=Local"
	db, err := gorm.Open(mysql.Open(dsn), &gorm.Config{})
	if err != nil {
		panic("failed to connect database")
	}

	// Migrate the schema
	db.AutoMigrate(&Pipeline{})

	// Create
	db.Create(&Pipeline{Namespace: "admin", Name: "p2"})
}
```